lib/ecto/repo/queryable.ex

```diff
@@ -11,11 +11,13 @@ defmodule Ecto.Repo.Queryable do
   require Ecto.Query
 
   def transaction(adapter, repo, fun, opts) when is_function(fun, 0) do
-    adapter.transaction(repo, opts, fun)
+    data = Ecto.Registry.lookup(repo)
```

Comment:
I am not sure this is the way we want to go. This is imposing all adapters to go through the Ecto.Registry, which means that now all adapters need to register something when they are initialized. I would prefer if, as this change relates only to pools, we did not change the adapter API at all, but only the SQL adapters which are the ones who effectively use a pool.